### PR TITLE
tables use tablehelpers.run

### PR DESCRIPTION
**Pull request by James-Pickett**
relates to https://github.com/kolide/launcher/issues/1321

Adds helper functions to enforce timeout and simplify creating / execing commands. Follow on PR will add linting.

```go
func RunSimple(ctx context.Context, slogger *slog.Logger, timeoutSeconds int, cmd allowedcmd.AllowedCommand, args []string, opts ...ExecOps)

func Run(ctx context.Context, slogger *slog.Logger, timeoutSeconds int, execCmd allowedcmd.AllowedCommand, args []string, stdout io.Writer, stderr io.Writer, opts ...ExecOps) error
```

### CODE REVIEW

File: pkg/osquery/table/touchid_system_darwin.go [hunk 1]
```diff
@@ -39,44 +38,35 @@ func (t *touchIDSystemConfigTable) generate(ctx context.Context, queryContext ta
 	var results []map[string]string
 	var touchIDCompatible, secureEnclaveCPU, touchIDEnabled, touchIDUnlock string
 
-	// Read the security chip from system_profiler
-	var stdout bytes.Buffer
-	cmd, err := allowedcmd.SystemProfiler(ctx, "SPiBridgeDataType")
```
Comment (directionless) [inline pkg/osquery/table/touchid_system_darwin.go:44]: There's a `context.WithTimeout` we can probably get rid of

**Review by directionless — COMMENTED** [review]
love it

**Review by RebeccaMahany — APPROVED** [review]
LGTM, just one small/optional question!

File: pkg/osquery/table/touchid_system_darwin.go [hunk 2]
```diff
@@ -33,50 +31,38 @@ type touchIDSystemConfigTable struct {
...
-	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
-	defer cancel()
-
 	var results []map[string]string
 	var touchIDCompatible, secureEnclaveCPU, touchIDEnabled, touchIDUnlock string
 
-	// Read the security chip from system_profiler
-	var stdout bytes.Buffer
-	cmd, err := allowedcmd.SystemProfiler(ctx, "SPiBridgeDataType")
+	stdout, err := tablehelpers.Exec(ctx, t.slogger, 15, allowedcmd.SystemProfiler, []string{"SPiBridgeDataType"}, false)
```
[the remark this answers is not in the document]
Comment (James-Pickett) [reply pkg/osquery/table/touchid_system_darwin.go:37]: woops.... I just missed using the same timeout and chose an arbitrary number, I'll update

[the remark this answers is not in the document]
Comment (directionless) [reply pkg/osquery/table/touchid_system_darwin.go:37]: The old 10s was also arbitrary.

[the remark this answers is not in the document]
Comment (directionless) [reply pkg/osquery/table/touchid_system_darwin.go:37]: If we want a default, we should have a default and drop it from the signature. Add it to Options. But I'm not sure yet

File: ee/tables/dism_default_associations/dism_default_associations.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 //go:build windows
 // +build windows
 
-package dsim_default_associations
+package dism_default_associations
```
Comment (James-Pickett) [inline ee/tables/dism_default_associations/dism_default_associations.go:4]: I corrected this spelling, then later remembered Becca having the same issue with it. Should I just put it back?

File: ee/tables/firmwarepasswd/firmwarepasswd.go [hunk 4]
```diff
@@ -90,42 +90,29 @@ func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) (
...
-func (t *Table) runFirmwarepasswd(ctx context.Context, subcommand string, output *bytes.Buffer) error {
-	ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
-	defer cancel()
-
-	cmd, err := allowedcmd.Firmwarepasswd(ctx, subcommand)
-	if err != nil {
-		return fmt.Errorf("creating command: %w", err)
-	}
-
+func (t *Table) runFirmwarepasswd(ctx context.Context, subcommand string) ([]byte, error) {
```
Comment (James-Pickett) [inline ee/tables/firmwarepasswd/firmwarepasswd.go:93]: this is not quite the same as it was, should probably leave it as close as possible for a large refactor

File: ee/tables/tablehelpers/exec.go [hunk 5]
```diff
@@ -19,66 +20,80 @@ import (
...
+// It enforces a timeout, logs, traces, and returns the stdout as a byte slice.
+// It discards stderr.
 //
-// `possibleBins` can be either a list of command names, or a list of paths to commands.
-// Where reasonable, `possibleBins` should be command names only, so that we can perform
-// lookup against PATH.
-func Exec(ctx context.Context, slogger *slog.Logger, timeoutSeconds int, execCmd allowedcmd.AllowedCommand, args []string, includeStderr bool, opts ...ExecOps) ([]byte, error) {
+// This is not suitable for high performance work -- it allocates new buffers each time
+// Use Run() for more control over the stdout and stderr streams.
+func RunSimple(ctx context.Context, slogger *slog.Logger, timeoutSeconds int, cmd allowedcmd.AllowedCommand, args []string, opts ...ExecOps) ([]byte, error) {
```
Comment (James-Pickett) [inline ee/tables/tablehelpers/exec.go:43]: still not 100% sold on the functions .... wonder if we should make an option for `WithStderr(w io.Writer)`

**Review by RebeccaMahany — DISMISSED** [review]

**Review by directionless — APPROVED** [review]
I think this looks good. I'm not sure how to test it, aside from checking a lot of tables.

File: ee/tables/tablehelpers/exec.go (hunk 5 above)
Comment (directionless) [reply ee/tables/tablehelpers/exec.go:43]: TBH I'm not 100% sure either. But I think this is a great step, and we can refactor if it feels wrong.

**Review by directionless — APPROVED** [review]
Still seems reasonable. Merge and iterate?

**Pull request merged**